cdap-support-bundle/bin/observability.py

```diff
@@ -0,0 +1,507 @@
+            "fileName": 'kubectl-exec-system-service-wrangler-status',
+            "fileExt": 'txt',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_status
+        }
+    ]
+
+    log_commands_processor(cdapCommandLists, directoryName, True, cdap_ns, pipeline_name, run_id)
+
+    update_progress(98)
+    compressed_file = shutil.make_archive(
+        base_name=directoryName,
+        format='gztar',
+        root_dir=parent_dir + directoryName
+    )
+
+    shutil.rmtree(directoryName)
+    update_progress(100)
+    print(f'Logs collection completed. Please find {directoryName}.tar.gz in current directory')
+    exit()
+
+
+def log_file_write(pathName, filename, fileWriteMode, contentToWrite, fileExt="log"):
+    f = open(f"{pathName}/{filename}.{fileExt}", fileWriteMode)
+    f.write(contentToWrite)
+    f.close()
+
+
+def update_progress(count):
+    total = 100
+    suffix = 'Done' if (count == 100) else 'In Progress'
+    bar_len = 60
+    filled_len = int(round(bar_len * count / float(total)))
+
+    percents = round(100.0 * count / float(total), 1)
+    bar = '=' * filled_len + '-' * (bar_len - filled_len)
+
+    sys.stdout.write('[%s] %s%s ...%s\r' % (bar, percents, '%', suffix))
+    sys.stdout.flush()
+
+
+def log_commands_processor(commandLists, directoryName, progressShow, cdap_ns, pipeline_name, run_id):
+    progress_len = 5
+    count = 1
+
+    for x in commandLists:
+        directory_name = directoryName
+        if "isPipelineLog" in x  and (cdap_ns is None or pipeline_name is None or run_id is None):
+            update_progress(progress_len)
+        else:
+            if x["parentFolder"] != "" and os.path.isdir(os.path.join(directoryName , x["parentFolder"])) == False:
+                os.makedirs(os.path.join(directoryName , x["parentFolder"]))
+
+            command_output = subprocess.getoutput(x["command"])
+            if x["fileWrite"] == True:
+                directory_name = os.path.join(directoryName , x["parentFolder"])
+                log_file_write(directory_name, f'{x["fileName"]}', "a", command_output, x["fileExt"])
+
+            if x["forEachNode"] == True:
+                nArr = command_output.split('\n')
+                for a in nArr:
+                    placeholderCommand = x["forEachCommand"]
+                    command = placeholderCommand.replace(x["placeholderString"], a)
+                    for_command_output = subprocess.getoutput(command)
+                    directory_name = os.path.join(directoryName , x["parentFolder"])
+                    log_file_write(directory_name , f'{x["childFileName"]}-{a}', "a", for_command_output, x["fileExt"])
+
+            if progressShow:
```

Comment:
progressShow is always True. Should we remove this param?

Reply:
done